web/src/unread_ops.js

```diff
@@ -160,7 +160,10 @@ function bulk_update_read_flags_for_narrow(narrow, op, args = {}) {
                     blueslip.log("Cleared old_unreads_missing after bankruptcy.");
                 }
             }
-            dialog_widget.close();
+
+            if (dialog_widget.is_open()) {
+                dialog_widget.close();
+            }
```

Comment:
Hmm, it might be better if we can have the callers pass the specific dialog that initiated this request? Given that this is happening after a network delay, the user might be using some totally different dialog by the time we get here.

Also, notably, is it intended that this happens even on recursive calls?

Reply:
> Hmm, it might be better if we can have the callers pass the specific dialog that initiated this request? 

Okay, did that.

> Also, notably, is it intended that this happens even on recursive calls?

Nice catch, fixed that too.